cpp/src/utilities/cython.cpp

```diff
@@ -27,28 +28,161 @@ namespace cython {
 void populate_graph_container(graph_container_t& graph_container,
-                              legacyGraphTypeEnum legacyType,
-                              raft::handle_t const& handle,
-                              void* offsets,
-                              void* indices,
+                              raft::handle_t& handle,
+                              void* src_vertices,
+                              void* dst_vertices,
                               void* weights,
-                              numberTypeEnum offsetType,
-                              numberTypeEnum indexType,
+                              void* vertex_partition_offsets,
+                              numberTypeEnum vertexType,
+                              numberTypeEnum edgeType,
                               numberTypeEnum weightType,
-                              int num_vertices,
-                              int num_edges,
-                              int* local_vertices,
-                              int* local_edges,
-                              int* local_offsets,
+                              int num_partition_edges,
+                              size_t num_global_vertices,
+                              size_t num_global_edges,
+                              size_t row_comm_size,  // pcols
+                              size_t col_comm_size,  // prows
                               bool transposed,
                               bool multi_gpu)
 {
   CUGRAPH_EXPECTS(graph_container.graph_ptr_type == graphTypeEnum::null,
                   "populate_graph_container() can only be called on an empty container.");
 
+  bool do_expensive_check{false};
+  bool hypergraph_partitioned{false};
+
+  // FIXME: Consider setting up the subcomms right after initializing comms, no
+  // need to delay to this point.
+  // Setup the subcommunicators needed for this partition on the handle.
+  partition_2d::subcomm_factory_t<partition_2d::key_naming_t, int> subcomm_factory(handle,
+                                                                                   row_comm_size);
+
+  // FIXME: once the subcomms are set up earlier (outside this function), remove
+  // the row/col_comm_size params and retrieve them from the handle (commented
+  // out lines below)
+  auto& row_comm           = handle.get_subcomm(cugraph::partition_2d::key_naming_t().row_name());
+  auto const row_comm_rank = row_comm.get_rank();
+  // auto const row_comm_size = row_comm.get_size(); // pcols
+  auto& col_comm           = handle.get_subcomm(cugraph::partition_2d::key_naming_t().col_name());
+  auto const col_comm_rank = col_comm.get_rank();
+  // auto const col_comm_size = col_comm.get_size(); // prows
+
+  // Copy the contents of the vertex_partition_offsets (host array) to a vector
+  // as needed by the partition_t ctor.
+  int* vertex_partition_offsets_array = reinterpret_cast<int*>(vertex_partition_offsets);
+
+  // FIXME: this needs to be vertex_t, not int?
```

Comment:
Yes, this needs to be vertex_t.

Reply:
To be addressed in PR #1178

Reply:
The `create_graph()` function in PR #1178 uses `vertex_t` for this.